### PR TITLE
Refresh of parallel iterators primer

### LINKED ISSUE #6343: Refresh parIter.chpl primer
Looking at the parIter primer while looking at issue #6337, it seems to me that we should improve it in the following ways:

* describe the definition of standalone iterators prior to leader-follower.  They're strictly simpler to write and understand, and more likely to be understood and defined by someone trying to write their first parallel iterator.

* clearly state the current requirement that to resolve parallel iterators, the user must provide a serial iterator as well (something that I hope we'll be able to relax over time -- see issue #6339).

There may be other things that should be improved as well -- these were just two that jumped out at me.  (edit: See also the notes in issue #8364).

### PR DESCRIPTION
**Pull request by bradcray**
This PR does a few things with the parallel iterators primer:

* makes some minor style and documentation tweaks to the `computeChunk()` utility routine as a follow on to issue #16890 that felt too nitpicky to bother with during the review process
* takes a full pass over the full parIters primer while here, to define standalone parallel iterators before leader-follower iterators (because they're simpler to understand and arguably a more common case to want to support).

Resolves #6343.

### CODE REVIEW

File: test/release/examples/primers/parIters.chpl [hunk 1]
```diff
@@ -173,54 +108,216 @@ writeln();
...
+
+Leader-follower Roles
+---------------------
+*/
+
+// At a high level, the role of a leader iterator is to:
+//
+//   a) create the parallel tasks used to implement the forall loop,
+//   b) associate the parallel tasks with specific locales as required/desired.
+//   c) assign work (e.g., iterations) to each parallel task
```
[the remark this answers is not in the document]
Comment (bradcray) [reply test/release/examples/primers/parIters.chpl:261]: Huh, I view `coforall l in Locales do on l` as a followed by b (though in practice our implementation optimizes by doing them simultaneously).

File: test/release/examples/primers/parIters.chpl [hunk 2]
```diff
@@ -1,152 +1,87 @@
...
-  Iterators in Chapel`_. We expect the parallel iterator interface
-  to change and improve over time, so this should be considered a
-  snapshot of their use in the current implementation.
+  This primer explains how to write parallel iterators in Chapel,
+  which can be used to drive parallel ``forall`` loops.  It assumes
+  that the reader already knows how to define serial iterators in
+  Chapel, as summarized in the iterators primer, for example.
+
+  Chapel has two main flavors of parallel iterators: `Standalone`
+  parallel iterators are the simplest form and can be used to define
```
[the remark this answers is not in the document]
Comment (bradcray) [reply test/release/examples/primers/parIters.chpl:10]: Good point.

File: test/release/examples/primers/parIters.chpl [hunk 3]
```diff
@@ -1,152 +1,87 @@
...
-
-  For a more thorough introduction to leader-follower iterators,
-  refer to our PGAS 2011 paper, `User-Defined Parallel Zippered
-  Iterators in Chapel`_. We expect the parallel iterator interface
-  to change and improve over time, so this should be considered a
-  snapshot of their use in the current implementation.
+  This primer explains how to write parallel iterators in Chapel,
+  which can be used to drive parallel ``forall`` loops.  It assumes
+  that the reader already knows how to define serial iterators in
+  Chapel, as summarized in the iterators primer, for example.
```
[the remark this answers is not in the document]
Comment (bradcray) [reply test/release/examples/primers/parIters.chpl:7]: Yeah, I was too lazy to figure out how to do this, but have done it now (see next commit).

**Comment by bradcray** [comment]
@e-kayrakli : I've pushed a new commit to address your two main points and mildly adjust the wording on the a, b, c list (which predates this PR and just moved around as a result of it).  Were you still in the process of reviewing this?

**Review by e-kayrakli — APPROVED** [review]

**Pull request merged**